docs/DeveloperGuide.md

```diff
@@ -133,6 +133,27 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add Student Feature
+
+#### Implementation
+The add student feature allows user to add a student to the TutorBuddy Application. 
+
+This feature makes use of `AddStudentCommandParser` and `AddStudentCommand` to create a new `Student` object. The operation can be accessed in the Model interface through `Model#addStudent()`. 
+
+Given below is an example of how the add student mechanism runs:
+1. The user executes the add student command with the command word `add_student` and include all the information required.
+2. `LogicManager` starts executing and parses the command using `addressBookParser`.
+3. `AddressBookParser` recognises the command and runs the `AddStudentCommandParser` class.
+4. `AddStudentCommandParser` then validates the information, and creates a new `Student` object.
+5. `AddStudentCommandParser` also creates a new `AddStudentCommand`.
+6. The command is returned to `LogicManager` which then executes the command.
```

Comment:
Hmm, since you mentioned that the `AddStudentCommandParser` creates both a `Student` and a `AddStudentCommand`, would it be better to also mention that the `AddStudentCommand` executes with the `Student` object?

Reply:
Took the comment and added the information